cloud/amazon/iam.py

```diff
@@ -133,7 +133,7 @@
     iam_type: user
     name: jdavila
     state: update
-    groups: "{{ item.created_group.group_name }}"
+    group: "{{ item.created_group.group_name }}"
```

Comment:
The original commit I had had the groups and not group... it's weird it got reverted...

The other changes are to fix a bug that caused the delete of a group to work.  Below are the fixes for that